main/kotlin/com/example/people_managment_system/controller/SystemController.kt

```diff
@@ -0,0 +1,52 @@
+package com.example.people_managment_system.controller
+
+import com.example.people_managment_system.dtos.EmployeeDto
+import com.example.people_managment_system.service.SystemService
+import org.springframework.dao.DataIntegrityViolationException
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+
+@RestController
+@RequestMapping("api/system")

```

Comment:
Good work on the controller. This controller has different operations to do with Company, you can split it by creating two controllers.

One can be `EmployeeController` and the other can be `CompanyController`.